### PR TITLE
Adding necessary plugins and config for pluggable scm jenkinsfile

### PR DESCRIPTION
**Pull request by lykabernardo**
* Adding the following plugins: pipeline-maven:3.5.8, cucumber:0.0.2, tasks:4.52, junit-attachments:1.5
* Updating sonar scanner plugin from 2.4.4 to 2.7.1
* Updating adop_sonar.groovy script for sonar scanner 2.7.1 configuration.

### CODE REVIEW

**Comment by nickdgriffin** [comment]
I think the Sonar changes need impacting please, any cartridge which depends on the version that currently exists probably won't load the Sonar section or could fail entirely. Perhaps we need to have two named runners that use the same version.

**Comment by lykabernardo** [comment]
I found two cartridge with Sonar configuration:
[adop-cartridge-java](https://github.com/Accenture/adop-cartridge-java/blob/master/jenkins/jobs/dsl/java_reference_application_jobs.groovy#L156-L170)
[adop-cartridge-doa-m3](https://github.com/Accenture/adop-cartridge-doa-m3/blob/master/jenkins/jobs/dsl/java_reference_application_jobs.groovy#L154-L168)

The sonar section of the analysis job is configured with sonar 2.2.1 version but still works with sonar 2.4.4 that we are currently using: https://github.com/Accenture/adop-jenkins/blob/master/resources/plugins.txt#L113.

I have tested loading and running these cartridge with sonar 2.7.1 (the version needed for jenkinsfile), the sonar section of the code analysis job of the cartridge were still loaded. The job still runs successfully with result.

File: resources/init.groovy.d/adop_sonar.groovy [hunk 1]
```diff
@@ -75,7 +63,7 @@ Thread.start {
 
     def sonarRunnerInstaller = new SonarRunnerInstaller(sonar_runner_version)
     def installSourceProperty = new InstallSourceProperty([sonarRunnerInstaller])
-    def sonarRunner_inst = new SonarRunnerInstallation("ADOP SonarRunner " + sonar_runner_version, "", [installSourceProperty])
+    def sonarRunner_inst = new SonarRunnerInstallation("Sonar_Scanner", "", [installSourceProperty])
```
Comment (dsingh07) [inline resources/init.groovy.d/adop_sonar.groovy]: Can we keep the name in the previous format i.e. ```ADOP SonarScanner```?

**Comment by lykabernardo** [comment]
@dsingh07 - updated the name to "ADOP SonarScanner".

**Comment by dsingh07** [comment]
Thanks @lykabernardo LGTM!

**Pull request merged**